Sources/LeafKit/DocumentLoader.swift

```diff
@@ -0,0 +1,204 @@
+struct UnresolvedDocument {
+    let name: String
+    let raw: [Syntax]
+    
+    var unresolvedDependencies: [String] {
+        return extensions.map { $0.key }
+    }
+    
+    private var extensions: [Syntax.Extend] {
+        return raw.compactMap {
+            switch $0 {
+            case .extend(let e): return e
+            default: return nil
+            }
+        }
+    }
+}
+
+struct ResolvedDocument {
+    let name: String
+    let ast: [Syntax]
+    
+    init(name: String, ast: [Syntax]) throws {
+        for syntax in ast {
+            switch syntax {
+            case .extend, .export:
+                // all extentions MUST be resolved
+                throw "unresolved ast"
+            default:
+                continue
+            }
+        }
+        
+        self.name = name
+        self.ast = ast
+    }
+}
+
+final class DocumentLoader {
```

Comment:
DocumentLoader's API seemed like it is more designed for the use cases required by the tests. But, maybe that's fine if it ends up being internal anyway. As long as the renderer type we end up with has an API like:

```swift
final class LeafRenderer {
    let config: LeafConfig
    let eventLoop: EventLoop

    func render<T>(_ path: String, _ context: T) -> EventLoopFuture<LeafView> 
        where T: Encodable 
    { ... }
}
```

Reply:
(I also think that DocumentLoader should be a struct)

Reply:
yea, I'm still really sorting out the best way to go up the fields to the rendering and what layer does what.. this is mostly just a way to have an object that can cache and load files.. maybe it gets absorbed by the renderer later, or stays internal

Reply:
keeping this for testing as mentioned, and moving work to renderer